lib/Service/IgnoreService.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace OCA\Recognize\Service;
+
+use OC\Files\Cache\CacheQueryBuilder;
+use OC\SystemConfig;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IDBConnection;
+use Psr\Log\LoggerInterface;
+
+class IgnoreService {
+	private IDBConnection $db;
+	private SystemConfig $systemConfig;
+	private LoggerInterface $logger;
+
+	public function __construct(IDBConnection $db, SystemConfig $systemConfig, LoggerInterface $logger) {
+		$this->db = $db;
+		$this->systemConfig = $systemConfig;
+		$this->logger = $logger;
+	}
+	/**
+	 * @param int $storageId
+	 * @param array $ignoreMarkers
+	 * @return list<string>
+	 */
+	public function getIgnoredDirectories(int $storageId, array $ignoreMarkers): array {
+		$qb = new CacheQueryBuilder($this->db, $this->systemConfig, $this->logger);
```

Comment:
What is CacheQueryBuilder?
Is the call to new necessary? (`$OC::server->get(CacheQueryBuilder::class)` should work?)

Applications should not use stuff from `OC`

Reply:
CacheQueryBuilder is an enhanced QueryBuilder class to access the file cache table. There is no OCP API for this, sadly, but we need to use the filecache table for performance reasons. We also need to create a new CacheQueryBuilder for each query, so I think $OC::server->get doesn't work.